### PR TITLE
New approach to visitor to allow mixed xxx-fix

### PR DESCRIPTION
**Pull request by caldwecr**
0.3.x

Worked with @mattvanhorn to redesign how visitation works from the ground up. 
Adds the aggregations visitor for faceted search.

### CODE REVIEW

File: lib/msfl_visitors/nodes/equal.rb [hunk 1]
```diff
@@ -2,6 +2,23 @@
 module MSFLVisitors
   module Nodes
     class Equal < Comparison
+
+      def accept(visitor)
+        visitor.visit self
+        #
```
Comment (mattvanhorn) [inline lib/msfl_visitors/nodes/equal.rb]: should probably get rid of this commented out code

File: spec/parsers/msfl_parser_spec.rb [hunk 2]
```diff
@@ -110,7 +110,7 @@
         # { value: { in: [50, 250, 20000] } }
```
Comment (mattvanhorn) [inline spec/parsers/msfl_parser_spec.rb]: can we lose this, too?

**Comment by mattvanhorn** [comment]
Other than cleaning up the commented-out stuff, (and maybe running this through rubocop) - it looks good to me.

**Comment by cymen** [comment]
:+1:

File: README.md [hunk 3]
```diff
@@ -1,27 +1,82 @@
...
- MSFLVisitors::AST.new(filter).accept(visitor)
+MSFLVisitors.get_chewy_clauses dataset, filter
 
- => 'make == "Toyota"'
+=> [{:clause=>"make == \"Toyota\""}]
 
- ```
+```
+
+## Facted example
```
Comment (mattvanhorn) [inline README.md]: Faceted?

**Pull request merged**